sashimi-webapp/src/database/data-modifier/dataAdd.js

```diff
@@ -1,26 +1,82 @@
-import exceptions from '../exceptions';
+import exceptions from 'src/database/exceptions';
+import constants from 'src/database/constants';
+import SqlCommands from 'src/database/sql-related/sqlCommands';
+import DateTime from 'src/database/generated-data/dateTime';
+import SqlArray from 'src/database/generated-data/sqlArray';
+import StringManipulator from 'src/database/stringManipulation';
 
-import constants from '../constants';
+const sqlCommands = new SqlCommands();
+const dateTime = new DateTime();
+const alasqlArray = new SqlArray();
+const stringManipulator = new StringManipulator();
 
-import SqlCommands from '../sql-related/sqlCommands';
+function isLowerUpperBoundWithinRange(lowerBound, upperBound, givenArray) {
+  return lowerBound < upperBound
+    && lowerBound >= 0
+    && upperBound < givenArray.length;
+}
 
-import DateTime from '../generated-data/dateTime';
+function exchange(givenArray, currIndex, thenIndex) {
+  const tempObject = givenArray[currIndex];
+  givenArray[currIndex] = givenArray[thenIndex];
+  givenArray[thenIndex] = tempObject;
+}
 
-import SqlArray from '../generated-data/sqlArray';
+function bubbleUp(givenArray, lowerBound, upperBound) {
+  if (isLowerUpperBoundWithinRange(lowerBound, upperBound, givenArray)) {
+    let prev = lowerBound;
+    for (let index = lowerBound + 1; index <= upperBound; index+=1) {
+      exchange(givenArray, prev, index);
+      prev = index;
+    }
+  }
+}
 
-const sqlCommands = new SqlCommands();
-const dateTime = new DateTime();
-const alasqlArray = new SqlArray();
+function generateUniqueNewFileName(queryFiles, defaultFileName) {
+  let newFileName = defaultFileName;
+  let duplicateCount = 0;
+  for (let fileIndex = 0; fileIndex < queryFiles.length; fileIndex+=1) {
+    const fileObject = queryFiles[fileIndex];
+    if (fileObject[constants.HEADER_FILE_MANAGER_FILE_NAME] === newFileName) {
+      duplicateCount += 1;
+      // generate new unique file name
+      newFileName = stringManipulator.stringConcat(
+      constants.DEFAULT_FILE_NAME_OMIT_FILE_TYPE, duplicateCount, '.md');
+    } else if (fileIndex < queryFiles.length) {
+      bubbleUp(queryFiles, fileIndex, queryFiles.length -1);
+      fileIndex -= 1;
+    }
+  }
+  return newFileName;
+}
```

Comment:
Because of the way `exactSearchStartFileNameInFolder` return its result, I suspect the `queryFiles` may contain something like this too?

```javascript
  { 'file_name': 'newFile-for-School.md' },
  { 'file_name': 'newFile-for-Work.md' },
  { 'file_name': 'newFile-for-Home.md' }
```

The algorithm will casue infinite loop too if the user have this kind of file structure, because it cannot find `newFile.md` and decrease the `fileIndex` by 1 every time.

```javascript
  { 'file_name': 'newFile01.md' },
  { 'file_name': 'newFile02.md' },
  { 'file_name': 'newFile03.md' }
```

```javascript
  { 'file_name': 'newFile' },
```